test/Triton/combine.mlir

```diff
@@ -1,27 +1,45 @@
 // RUN: triton-opt %s -split-input-file -canonicalize -triton-combine | FileCheck %s
 
-// CHECK-LABEL: @test_combine_dot_add_pattern
-tt.func @test_combine_dot_add_pattern() -> (tensor<128x128xf32>, tensor<128x128xf32>) {
-    // CHECK-DAG: %[[d:.*]] = arith.constant dense<3.000000e+00> : tensor<128x128xf32>
-    // CHECK-DAG: %[[b:.*]] = arith.constant dense<2.000000e+00> : tensor<128x128xf32>
-    // CHECK-DAG: %[[a:.*]] = arith.constant dense<1.000000e+00> : tensor<128x128xf32>
+// We don't combine if the dot result is used by more than one op.
+// CHECK-LABEL: @test_combine_dot_add_invalid_pattern
+tt.func @test_combine_dot_add_invalid_pattern() -> (tensor<128x128xf32>, tensor<128x128xf32>) {
     %a = arith.constant dense<1.0> : tensor<128x128xf32>
     %b = arith.constant dense<2.0> : tensor<128x128xf32>
     %zero = arith.constant dense<0.0> : tensor<128x128xf32>
     %d = arith.constant dense<3.0> : tensor<128x128xf32>
 
     %dot_out = tt.dot %a, %b, %zero {allowTF32 = true, maxNumImpreciseAcc = 0 : i32, transA = false, transB = false} : tensor<128x128xf32> * tensor<128x128xf32> -> tensor<128x128xf32>
 
-    // CHECK-NEXT: %[[res0:.*]] = tt.dot %[[a]], %[[b]], %[[d]] {allowTF32 = true, maxNumImpreciseAcc = 0 : i32} : tensor<128x128xf32> * tensor<128x128xf32> -> tensor<128x128xf32>
+    // CHECK: arith.addf
     %res0 = arith.addf %dot_out, %d : tensor<128x128xf32>
 
-    // CHECK-NEXT: %[[res1:.*]] = tt.dot %[[a]], %[[b]], %[[d]] {allowTF32 = true, maxNumImpreciseAcc = 0 : i32} : tensor<128x128xf32> * tensor<128x128xf32> -> tensor<128x128xf32>
+    // CHECK-NEXT: arith.addf
```

Comment:
OOC, both `addf` in here have the same operands, shouldn't we still have combined and removed the duplicate `dot` ?

Reply:
I think we should expect that CSE has ran by that time, but maybe we want to change the test to be less confusing

Reply:
Indeed good suggestions. Thank you and fixed.